### PR TITLE
Add REdit implementation details in DG

### LINKED ISSUE #269: Add implementation details for review edit in Developer Guide

### PR DESCRIPTION
**Pull request by MinHeinA**
Resolve #269

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -209,9 +209,52 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 3. Since the user input is valid, the `AddressBookParser` will create a `REditCommandParser` to parse the command arguments, `2 r/5`.
+
+Step 4. `REditCommandParser` will parse the index to a `Index` object and parse other arguments as a `EditReviewDescriptor` object. The `Index` and `EditReviewDescriptor` objects will then be passed to the returned `REditCommand` object as its arguments.
+
+Step 5. In `LogicManager`, the returned `REditCommand` is executed.
+
+Step 6. A new `Review` object is created with the fields of `Review` to be edited, except the field(s) that are to be edited, which will be set by the parsed user input. In this case `Review` with index 2 is copied over to a new `Review` object expect its `Rating` field which is set as `5`.
+
+Step 7. The new `Review` object will then replace the `Review` object to be edited in the `Model` when `model.setReview()` is called.
```
Comment (clarence-chew) [inline docs/DeveloperGuide.md:244]: This is not very accurate -
1. It finds the `Stall` attached to the `Review` and replaces the `Stall` with an edited copy containing the new review
2. It updates its `UniqueReviewList` to contain the `Review`s of all the `Stall`s

Maybe a better wording is that "`model.setReview()` will replace the `Stall` containing that review, and the `Review`".

**Review by clarence-chew — COMMENTED** [review]
Due to how the model handles respecting invariants in Stalls and Reviews, the methods modifying the stalls and reviews aren't very straightforward.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (MinHeinA) [reply docs/DeveloperGuide.md]: Implementation updated with the details above. For further review by @clarence-chew

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -269,6 +269,52 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 3. Since the user input is valid, the `AddressBookParser` will create a `REditCommandParser` to parse the command arguments, `2 r/5`.
+
+Step 4. `REditCommandParser` will parse the index to a `Index` object and parse other arguments as a `EditReviewDescriptor` object. The `Index` and `EditReviewDescriptor` objects will then be passed to the returned `REditCommand` object as its arguments.
+
+Step 5. In `LogicManager`, the returned `REditCommand` is executed.
+
+Step 6. A new `Stall` object is created with the fields of `Review` to be edited, except the field(s) that are to be edited, which will be set by the parsed user input. In this case `Review` with index 2 is copied over to a new `Review` object expect its `Rating` field which is set as `5`.
+
+Step 6. The `Stall` which contains the `Review` to be edited is copied to a new `Stall` object with all of its data fields except the `Review` to be edited which will be set by the parsed user input. In this case `Review` with index 2 is copied over to a new `Review` object expect its `Rating` field which is set as `5`.
```
Comment (clarence-chew) [inline docs/DeveloperGuide.md:301]: Repeated step 6.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -269,6 +269,52 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 2. The user executes `redit 2 r/5` command to edit `Review` with index 2 to edit its rating to 5.
+
+Step 3. Since the user input is valid, the `AddressBookParser` will create a `REditCommandParser` to parse the command arguments, `2 r/5`.
+
+Step 4. `REditCommandParser` will parse the index to a `Index` object and parse other arguments as a `EditReviewDescriptor` object. The `Index` and `EditReviewDescriptor` objects will then be passed to the returned `REditCommand` object as its arguments.
+
+Step 5. In `LogicManager`, the returned `REditCommand` is executed.
+
+Step 6. A new `Stall` object is created with the fields of `Review` to be edited, except the field(s) that are to be edited, which will be set by the parsed user input. In this case `Review` with index 2 is copied over to a new `Review` object expect its `Rating` field which is set as `5`.
```
Comment (clarence-chew) [inline docs/DeveloperGuide.md:299]: I think this is inaccurate - when model.setReview() called, then the Model handles making of the new Stall object.

**Review by clarence-chew — COMMENTED** [review]
I have some concerns.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (clarence-chew) [reply docs/DeveloperGuide.md]: I'm sorry for being a bit vague in this, the Model finds the Stall attached to the Review, not in the execution of the REditCommand instance

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -269,6 +269,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 3. Since the user input is valid, the `AddressBookParser` will create a `REditCommandParser` to parse the command arguments, `2 r/5`.
+
+Step 4. `REditCommandParser` will parse the index to a `Index` object and parse other arguments as a `EditReviewDescriptor` object. The `Index` and `EditReviewDescriptor` objects will then be passed to the returned `REditCommand` object as its arguments.
+
+Step 5. In `LogicManager`, the returned `REditCommand` is executed.
+
+Step 6. A new `Review` object is created with the fields of `Review` to be edited, except the field(s) that are to be edited, which will be set by the parsed user input. In this case `Review` with index 2 is copied over to a new `Review` object expect its `Rating` field which is set as `5`.
+
+Step 7. A new `Stall` object will be created and its field copied over except the `Review` to be edited which will be replaced by the updated `Review`. This new `Stall` object will then replace the `Stall` object that contains the `Review` to be edited in the `Model` when `model.setReview()` is called.
```
Comment (clarence-chew) [inline docs/DeveloperGuide.md:301]:
```
Step 6. In the execution of the `REditCommand`, a new `Review` object is created. This `Review` consists of the field(s) from the parsed user input, replacing some field(s) of the original `Review` object. In this case, the original `Review` with index 2 is copied over to a new `Review` object except its `Rating` field which is set as `5`.

Step 7. `model.setReview()` will interact with the model to have it replace the immutable `Stall` which contained the original `Review` with a `Stall` containing the updated `Review`.
```

Comment (MinHeinA) [reply docs/DeveloperGuide.md:301]: Updated docs as per recommendation. Thank you.

**Review by clarence-chew — APPROVED** [review]
LGTM!

**Pull request merged**